src/assoc.js

```diff
@@ -22,6 +23,16 @@ var _curry3 = require('./internal/_curry3');
  *      R.assoc('c', 3, {a: 1, b: 2}); //=> {a: 1, b: 2, c: 3}
  */
 module.exports = _curry3(function assoc(prop, val, obj) {
+  if (_isArray(obj)) {
+    var newArr = [].concat(obj);
+    if (prop < 0 && Math.abs(prop) > newArr.length) {
+      throw new Error('Invalid index');
```

Comment:
Also, this should be a [RangeError][1].


[1]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/RangeError